### PR TITLE
Stream.list can filter by subject

### PR DESCRIPTION
**Pull request by brandynbennett**
Helpful when we need to find which stream belongs to a subject

### CODE REVIEW

**Review by marmor157 — APPROVED** [review]

File: lib/jetstream/api/stream.ex [hunk 1]
```diff
@@ -345,17 +345,23 @@ defmodule Jetstream.API.Stream do
   @doc """
   Paged list of known Streams including all their current information.
 
+  ## Options
+
+  * `:offset`
+  * `:subject` - A subject the Stream must collect to appear in the list
```
Comment (mkaput) [inline lib/jetstream/api/stream.ex:351]:
```suggestion
  * `:subject` - A subject the `Stream` must collect to appear in the list.
```

File: lib/jetstream/api/stream.ex [hunk 2]
```diff
@@ -345,17 +345,23 @@ defmodule Jetstream.API.Stream do
   @doc """
   Paged list of known Streams including all their current information.
 
+  ## Options
+
+  * `:offset`
```
Comment (mkaput) [inline lib/jetstream/api/stream.ex:350]: It would be great that this option is used for pagination.

Question: what about `limit`?

**Review by mkaput — CHANGES_REQUESTED** [review]

File: lib/jetstream/api/stream.ex (hunk 2 above)
Comment (brandynbennett) [reply lib/jetstream/api/stream.ex]: Good question. I gave this a try and it looks like the nats server API doesn't accept `limit`.

Comment (mkaput) [reply lib/jetstream/api/stream.ex]: Ok

**Pull request merged**